Framework/API/src/FunctionFactory.cpp

```diff
@@ -361,6 +363,23 @@ void FunctionFactoryImpl::addTie(IFunction_sptr fun,
   }
 }
 
+std::vector<std::string> FunctionFactoryImpl::getFunctionNamesGUI() const {
+  auto &allNames = getFunctionNames<IFunction1D>();
+  std::vector<std::string> names;
+  names.reserve(allNames.size());
+  auto excludes =
+      Kernel::ConfigService::Instance().getString("curvefitting.guiExclude");
+  Kernel::StringTokenizer tokenizer(excludes, ";",
+                                    Kernel::StringTokenizer::TOK_TRIM);
+  std::set<std::string> excludeList(tokenizer.begin(), tokenizer.end());
+  for (auto &name : allNames) {
```

Comment:
I was just fixing some `cppcheck` 1.87 warnings, and here I think it would tell one to use `std::copy_if` instead of the raw loop. Maybe it is worth trying?
```c++
std::copy_if(allNames.cbegin(), allNames.cend(), std::back_inserter(names), [&excludeList](const auto &name) {
  return excludedList.count(name) == 0
});
```